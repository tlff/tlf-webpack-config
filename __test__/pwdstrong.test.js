import App from "../src/js/component/pwdstrong.vue";
import { mount } from "vue-test-utils";
import Vue from "VUE";

describe("reset.test.js", () => {
    let cmp, vm;

    beforeEach(() => {
        document.body.innerHTML = `
            <div id="reset_form"></div>
        `;
        jsdom.reconfigure({
            url: "http://localhost/index.php?/reset&token=asdf"
        });
    });

    it('测试密码 a82820070 是中等密码', () => {
        cmp = mount(App, {
            propsData: {
                value: "a82820070"
            }
        })
        expect(cmp.vm.value).toBe("a82820070");
        expect(cmp.vm.color).toEqual({
            Lcolor:"#ffd8b4",
            Mcolor:"#ffaf56",
            Hcolor:"#eeeeee"
        });
        expect(cmp.vm.$el).toMatchSnapshot();
    });
    test("测试 12$#afiojgSF 是复杂密码", () => {
        cmp = mount(App, {
            propsData: {
                value: "12$#afiojgSF"
            }
        });
        expect(cmp.vm.color).toEqual({
            Lcolor:"#ffd8b4",
            Mcolor:"#ffaf56",
            Hcolor:"#e85959"
        });
        expect(cmp.vm.value).toBe("12$#afiojgSF");
        expect(cmp.vm.$el).toMatchSnapshot();
    });
    test("测试 1234123 是弱密码", () => {
        cmp = mount(App, {
            propsData: {
                value: "1234123"
            }
        })
        expect(cmp.vm.value).toBe("1234123");
        expect(cmp.vm.$el).toMatchSnapshot();
    })
    test("测试 '123' 是弱密码", () => {
        cmp = mount(App, {
            propsData: {
                value: "123"
            }
        })
        expect(cmp.vm.value).toBe("123");
        expect(cmp.vm.$el).toMatchSnapshot();
    })
});