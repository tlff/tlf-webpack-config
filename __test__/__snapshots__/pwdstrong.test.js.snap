// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reset.test.js 测试 '123' 是弱密码 1`] = `
<ul
  class="pass_set"
>
  <li
    class="strength_L"
    style="background-color: rgb(238, 238, 238);"
  >
    
    弱
  
  </li>
   
  <li
    class="strength_M"
    style="background-color: rgb(238, 238, 238);"
  >
    
    中
  
  </li>
   
  <li
    class="strength_H"
    style="background-color: rgb(238, 238, 238);"
  >
    
    强
  
  </li>
</ul>
`;

exports[`reset.test.js 测试 12$#afiojgSF 是复杂密码 1`] = `
<ul
  class="pass_set"
>
  <li
    class="strength_L"
    style="background-color: rgb(255, 216, 180);"
  >
    
    弱
  
  </li>
   
  <li
    class="strength_M"
    style="background-color: rgb(255, 175, 86);"
  >
    
    中
  
  </li>
   
  <li
    class="strength_H"
    style="background-color: rgb(232, 89, 89);"
  >
    
    强
  
  </li>
</ul>
`;

exports[`reset.test.js 测试 1234123 是弱密码 1`] = `
<ul
  class="pass_set"
>
  <li
    class="strength_L"
    style="background-color: rgb(255, 216, 180);"
  >
    
    弱
  
  </li>
   
  <li
    class="strength_M"
    style="background-color: rgb(238, 238, 238);"
  >
    
    中
  
  </li>
   
  <li
    class="strength_H"
    style="background-color: rgb(238, 238, 238);"
  >
    
    强
  
  </li>
</ul>
`;

exports[`reset.test.js 测试密码 a82820070 是中等密码 1`] = `
<ul
  class="pass_set"
>
  <li
    class="strength_L"
    style="background-color: rgb(255, 216, 180);"
  >
    
    弱
  
  </li>
   
  <li
    class="strength_M"
    style="background-color: rgb(255, 175, 86);"
  >
    
    中
  
  </li>
   
  <li
    class="strength_H"
    style="background-color: rgb(238, 238, 238);"
  >
    
    强
  
  </li>
</ul>
`;
